File: ASWebUI-Angular-CLI/src/app/Models/sale-detail.model.ts
import { Identification } from "src/app/models/Identification.interface";
import { Product } from "./product.model"


export class SaleDetail implements Identification {
  id?: number;
  productId: number;
  invoiceId: number;
  productName: string;
  description: string;
  quantity: number;
  unitPrice: number;
  subTotal: number;
  total: number;
  tax?: number;
  
  constructor({
    id = 0,
    invoiceId = 0,
    productId=0,
    productName = "",
    description = "",
    quantity = 0,
    unitPrice = 0,
    subTotal = 0,
    total = 0,
    tax = 0,
  } = {}) {
    this.id = id;
    this.invoiceId = invoiceId;
    this.productId = productId;
    this.productName = productName;
    this.description = description;
    this.quantity = quantity;
    this.unitPrice = unitPrice;
    this.subTotal = subTotal;
    this.total = total;
    this.tax = tax;
  }
}

File: ASDataManager.Library/Models/SaleDetailModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ASDataManager.Library.Models
{
    public class SaleDetailModel
    {
        public int id { get; set; }
        public int InvoiceId { get; set; }
        public int ProductId { get; set; }
        public string ProductName { get; set; }
        public string Description { get; set; }
        public int UnitPrice { get; set; }
        public int Quantity { get; set; }
        public decimal Subtotal { get; set; }
        public decimal Tax { get; set; }
        public decimal Total { get; set; }


        public SaleDetailModel(int id, int invoiceId, int productId, string productName, string description, int purchasePrice, int quantity, decimal subtotal, decimal tax, decimal total)
        {
            this.id = id;
            InvoiceId = invoiceId;
            ProductId = productId;
            ProductName = productName;
            Description = description;
            UnitPrice = purchasePrice;
            Quantity = quantity;
            Subtotal = subtotal;
            Tax = tax;
            Total = total;
        }

        public SaleDetailModel() { }

    }
}